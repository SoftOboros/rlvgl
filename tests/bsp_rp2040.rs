//! Round-trip test for the RP2040 YAML adapter.
#![cfg(all(feature = "creator", feature = "regression"))]

#[path = "../src/bin/creator/bsp/ir.rs"]
mod ir;
#[path = "../src/bin/creator/bsp/rp2040.rs"]
mod rp2040;

use minijinja::{Environment, context};

#[test]
fn rp2040_roundtrip_snapshot() {
    let yaml = include_str!("fixtures/rp2040.yaml");
    let spec = rp2040::yaml_to_ir(yaml).unwrap();
    insta::assert_yaml_snapshot!("ir", &spec);

    let mut env = Environment::new();
    env.add_template(
        "gen",
        include_str!("../src/bin/creator/bsp/templates/simple.rs.jinja"),
    )
    .unwrap();
    let tmpl = env.get_template("gen").unwrap();
    let rendered = tmpl.render(context! { spec => &spec }).unwrap();
    insta::assert_snapshot!("generated", rendered);
}
