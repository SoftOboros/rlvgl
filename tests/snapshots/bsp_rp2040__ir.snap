---
source: tests/bsp_rp2040.rs
expression: "&spec"
---
mcu: RP2040
package: QFN56
clocks:
  pll: {}
  kernels: {}
pinctrl:
  - pin: GPIO0
    func: SPI0_SCK
    af: 0
  - pin: GPIO1
    func: SPI0_MOSI
    af: 0
peripherals:
  spi0:
    class: spi
    signals:
      sck: GPIO0
      mosi: GPIO1
